File: examples/python/calculator.py
"""Calculator module providing basic arithmetic operations."""


def add(a: int, b: int) -> int:
    """Return the sum of two integers."""
    return a + b


def subtract(a: int, b: int) -> int:
    """Return the difference of two integers."""
    return a - b


def multiply(a: int, b: int) -> int:
    """Return the product of two integers."""
    return a * b


def divide(a: int, b: int) -> float:
    """Return the quotient of two integers.
    
    Args:
        a: The dividend
        b: The divisor
        
    Returns:
        The quotient as a float
        
    Raises:
        ValueError: If b is zero
    """
    if b == 0:
        raise ValueError("Cannot divide by zero")
    return a / b

File: examples/rust/calculator.rs
//! Calculator module providing basic arithmetic operations.

/// Returns the sum of two integers.
pub fn add(a: i32, b: i32) -> i32 {
    a + b
}

/// Returns the difference of two integers.
pub fn subtract(a: i32, b: i32) -> i32 {
    a - b
}

/// Returns the product of two integers.
pub fn multiply(a: i32, b: i32) -> i32 {
    a * b
}

/// Returns the quotient of two integers.
/// 
/// # Arguments
/// * `a` - The dividend
/// * `b` - The divisor
/// 
/// # Returns
/// * `Ok(i32)` - The quotient
/// * `Err(String)` - Error if b is zero
pub fn divide(a: i32, b: i32) -> Result<i32, String> {
    if b == 0 {
        return Err("Cannot divide by zero".to_string());
    }
    Ok(a / b)
}
